src/node.cc

```diff
@@ -2834,6 +2837,13 @@ void SetupProcessObject(Environment* env,
     READONLY_PROPERTY(process, "traceDeprecation", True(env->isolate()));
   }
 
+  // --warn-on-sync
+  if (warn_on_sync) {
+    READONLY_PROPERTY(process, "warnOnSync", True(env->isolate()));
```

Comment:
though you're right. a PR to conditionally set those would be fairly trivial. i'll put that on my list, unless you get to it first. :)